### ISSUE TITLE
DCPU16AsmPrinter::printSrcMemOperand() looks outdated

### ISSUE THREAD

**Issue by ghost** [llvm-dcpu16/llvm-dcpu16#140]
Hi!

C-Code:

```
unsigned func(void) {
    unsigned x;
    asm volatile ("SET %0, SP": "=m" (x):);
    return x;
}
```

ASM:

```
    ; .file "test.c"
    .text
    .globl  func
    ; .align    1
:func
    SUB I, 0x1
    ;APP
    SET 0(I), SP
    ;NO_APP
    SET A, [I]
    ADD I, 0x1
    SET PC, POP

```

As you can see, the inline assembly will generate an not supported addressing mode. It looks like DCPU16AsmPrinter::printSrcMemOperand() is outdated and needs to be changed to print out something like `SET [0+I], SP`.

BTW: I know that  `"=r" (x)` will generate better code.

**Comment by ghost** [comment]
We should also fix the tests, which are mentioned in #85.

**Comment by ghost** [comment]
Fixed by 1c9e9e553f0b2014d766c68c516a1959d37557e6
